### PR TITLE
Clean up metric customization to get rid of envoyfilter based text editing

### PR DESCRIPTION
**Pull request by bianpengyuan**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by frankbu** [comment]
@bianpengyuan FYI, there is a test for this doc in #7944.

**Comment by bianpengyuan** [comment]
@frankbu Thanks for the link!

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 1]
```diff
@@ -23,15 +23,18 @@ By default, Istio defines and generates a set of standard metrics (e.g.
...
 `EnvoyFilter`: `definitions` and `metrics`. The `definitions` section
 supports creating new metrics by name, the expected value expression, and the
 metric type (`counter`, `gauge`, and `histogram`). The `metrics` section
 provides values for the metric dimensions as expressions, and allows you to
 remove or override the existing metric dimensions. You can modify the standard
-metric definitions using `tags_to_remove` or by re-defining a dimension.
+metric definitions using `tags_to_remove` or by re-defining a dimension. These
+configuration sections are also exposed as istioctl installation options, which
+allow you to customize metrics differently for gateway and sidecar at inbound
+or outbound direction.
```
Comment (frankbu) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:37]:
```suggestion
configuration settings are also exposed as istioctl installation options, which
allow you to customize different metrics for gateways and sidecars as well as
for the inbound or outbound direction.
```

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 2]
```diff
@@ -43,63 +46,71 @@ installation.
 
 ## Enable custom metrics
 
-Edit the `EnvoyFilter` to add or modify dimensions and metrics. Then, add
-annotations to all the Istio-enabled pods to extract the new or modified
-dimensions.
+1. By default, Telemetry v2 `EnvoyFilter` configuration is equivalent to the following installation options:
```
Comment (frankbu) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:49]:
```suggestion
1. The default telemetry v2 `EnvoyFilter` configuration is equivalent to the following installation options:
```

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 3]
```diff
@@ -43,63 +46,71 @@ installation.
...
+            inboundSidecar:
+              debug: false
+              stat_prefix: istio
+            outboundSidecar:
+              debug: false
+              stat_prefix: istio
+            gateway:
+              debug: false
+              stat_prefix: istio
+              disable_host_header_fallback: true
```
Comment (frankbu) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:69]:
```suggestion
    apiVersion: install.istio.io/v1alpha1
    kind: IstioOperator
    spec:
      value:
        telemetry:
          v2:
            prometheus:
              configOverride:
                inboundSidecar:
                  debug: false
                  stat_prefix: istio
                outboundSidecar:
                  debug: false
                  stat_prefix: istio
                gateway:
                  debug: false
                  stat_prefix: istio
                  disable_host_header_fallback: true
```

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 4]
```diff
@@ -43,63 +46,71 @@ installation.
...
+                    request_host: request.host
+            gateway:
+              debug: false
+              stat_prefix: istio
+              disable_host_header_fallback: true
+              metrics:
+                - name: requests_total
+                  dimensions:
+                    destination_port: string(destination.port)
+                    request_host: request.host
```
Comment (frankbu) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:113]:
```suggestion
    apiVersion: install.istio.io/v1alpha1
    kind: IstioOperator
    spec:
      value:
        telemetry:
          v2:
            prometheus:
              configOverride:
                inboundSidecar:
                  debug: false
                  stat_prefix: istio
                  metrics:
                    - name: requests_total
                      dimensions:
                        destination_port: string(destination.port)
                        request_host: request.host
                outboundSidecar:
                  debug: false
                  stat_prefix: istio
                  metrics:
                    - name: requests_total
                      dimensions:
                        destination_port: string(destination.port)
                        request_host: request.host
                gateway:
                  debug: false
                  stat_prefix: istio
                  disable_host_header_fallback: true
                  metrics:
                    - name: requests_total
                      dimensions:
                        destination_port: string(destination.port)
                        request_host: request.host
```

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 5]
```diff
@@ -43,63 +46,71 @@ installation.
...
     {{< /text >}}
 
-1. Save `stats-filter-1.6.yaml` and then apply the configuration using the following command:
+    To customize Telemetry v2 metrics, for example to add `request_host`
+    and `destionation_port` tags to `requests_total` metric emitted by gateway
+    and sidecars both at inbound and outbound direction, you can augment installation
+    options as following. Note you only need to specify the configuration override
+    section that you need customization. For example, if you only want to customize
+    sidecar inbound `requests_count` metric, you can omit `outboundSidecar` and
+    `inboundSidecar` sections in the following configuration.
```
Comment (frankbu) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:78]:
```suggestion
    To customize telemetry v2 metrics, for example, to add `request_host`
    and `destination_port` dimensions to the `requests_total` metric emitted by both
    gateways and sidecars in the inbound and outbound direction, change the installation
    options as follows:
    
    {{< tip >}}
    You only need to specify the configuration for the settings that you want to customize.
    For example, to only customize the sidecar inbound `requests_count` metric, you can omit
    in the `outboundSidecar` and `inboundSidecar` sections in the configuration. Unspecified
    settings will retain the default configuration, equivalent to the explicit settings shown above.
    {{< /tip >}}
```

**Pull request merged**